packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1178,11 +1178,14 @@ abstract class ResidentRunner extends ResidentHandlers {
     String route,
   });
 
+  /// The first time we do a run hot a full restart is not needed
```

Comment:
How about something like:

```
/// Connect to a flutter application.
///
/// [needsFullRestart] defaults to `true`, and controls if the frontend server should
/// compile a full dill. This should be set to `false` if this is called in [ResidentRunner.run], since that method already perfoms an initial compilation.

```